packages/ethereum-light-client/src/verify.rs

```diff
@@ -0,0 +1,396 @@
+    );
+
+    // There are two options to do a light client update:
+    // 1. We are updating the header with a newer one.
+    // 2. We haven't set the next sync committee yet and we can use any attested header within the same
+    // signature period to set the next sync committee. This means that the stored header could be larger.
+    // The light client implementation needs to take care of it.
+    ensure(
+        update_attested_slot > trusted_consensus_state.finalized_slot()
+            || (update_attested_period == stored_period
+                && update.next_sync_committee.is_some()
+                && trusted_consensus_state.next_sync_committee().is_none()),
+        EthereumIBCError::IrrelevantUpdate {
+            update_attested_slot,
+            trusted_finalized_slot: trusted_consensus_state.finalized_slot(),
+            update_attested_period,
+            stored_period,
+            update_sync_committee_is_set: update.next_sync_committee.is_some(),
+            trusted_next_sync_committee_is_set: trusted_consensus_state
+                .next_sync_committee()
+                .is_some(),
+        },
+    )?;
+
+    // Verify that the `finality_branch`, if present, confirms `finalized_header`
+    // to match the finalized checkpoint root saved in the state of `attested_header`.
+    is_valid_light_client_header(client_state, &update.finalized_header)?;
+    let finalized_root = update.finalized_header.beacon.tree_hash_root();
+
+    // This confirms that the `finalized_header` is really finalized.
+    validate_merkle_branch(
+        finalized_root,
+        update.finality_branch.clone().into(),
+        floorlog2(FINALIZED_ROOT_INDEX),
+        get_subtree_index(FINALIZED_ROOT_INDEX),
+        update.attested_header.beacon.state_root,
+    )
+    .map_err(|e| EthereumIBCError::ValidateFinalizedHeaderFailed(Box::new(e)))?;
+
+    // Verify that if the update contains the next sync committee, and the signature periods do match,
+    // next sync committees match too.
+    if let (Some(next_sync_committee), Some(stored_next_sync_committee)) = (
+        &update.next_sync_committee,
+        trusted_consensus_state.next_sync_committee(),
+    ) {
+        if update_attested_period == stored_period {
+            ensure(
+                next_sync_committee == stored_next_sync_committee,
+                EthereumIBCError::NextSyncCommitteeMismatch {
+                    expected: stored_next_sync_committee.aggregate_pubkey,
+                    found: next_sync_committee.aggregate_pubkey,
+                },
+            )?;
+        }
+        // This validates the given next sync committee against the attested header's state root.
+        validate_merkle_branch(
+            next_sync_committee.tree_hash_root(),
+            update
+                .next_sync_committee_branch
+                .clone()
+                .unwrap_or_default()
+                .into(),
+            floorlog2(NEXT_SYNC_COMMITTEE_INDEX),
```

Comment:
Don't you already have a constant for this `NEXT_SYNC_COMMITTEE_BRANCH_SIZE`

Reply:
Yup! Have moved them all and using them everywhere now.